### PR TITLE
Support for Google Gemini LLMs and Embeddings

### PR DESCRIPTION
**Pull request by uw4**
Initial support for Gemini, enables usage of Google LLMs and embedding models (see settings-gemini.yaml)

Install via
poetry install --extras "llms-gemini embeddings-gemini"

Notes:
* had to bump llama-index-core to later version that supports Gemini
* poetry --no-update did not work: Gemini/llama_index seem to require more (transient) updates to make it work...

### CODE REVIEW

**Comment by raphant** [comment]
Gemini's 1m context window for pro and flash will seem to be a great aid to ingesting and analyzing big documents. Looking forward to seeing this implemented!

File: pyproject.toml [hunk 1]
```diff
@@ -47,6 +49,9 @@ sentence-transformers = {version ="^2.6.1", optional = true}
 # Optional UI
 gradio = {version ="^4.19.2", optional = true}
 
+# Optional Google Gemini dependency
+google-generativeai = {version ="^0.5.4", optional = true}
```
Comment (jaluma) [inline pyproject.toml:53]: Why is it necessary to specify this package? It is already included in the `llama-index-llms-gemini` and `llama-index-embeddings-gemini`

Comment (jaluma) [reply pyproject.toml:53]: Also, can you fix mypy problems related to bump `llama-index`?

**Review by jaluma — APPROVED** [review]

**Comment by jaluma** [comment]
Update PR with last changes

**Pull request merged**